### ISSUE TITLE
Attachment : Xeroizer::InvoiceNotFoundError

### ISSUE THREAD

**Issue by maxime-d** [waynerobinson/xeroizer#234]
I'm trying to upload an attachment for an invoice but I get `Xeroizer::InvoiceNotFoundError: Invoice not found`.

Example : 

```ruby
xero.Invoice.attach_file(
  params[:invoice_guid],
  params[:file][:name],
  params[:file][:data],
  params[:file][:content_type]
)
```

or 

```ruby
invoice = xero.Invoice(params[:invoice_guid])
invoice.attach_file(
  params[:file][:name],
  params[:file][:data],
  params[:file][:content_type]
)
```

Anyone has a same issue ?

**Comment by stephaneGJ** [comment]
I have the same issue. Did you find any solution ? Thanks!

**Comment by nnc** [comment]
Try something like this:

```ruby
invoice = xero.Invoice.find(params[:invoice_guid])
invoice.attach_data(
  params[:file][:name],
  params[:file][:data],
  params[:file][:content_type]
)
```

It seems like you're passing the actual file data in `params[:file][:data]` so you need to use`attach_data` for that.
If you are actually passing in full path to the file, then `attach_file` is what you want.

Internally, `attach_file` just reads the whole file into memory and then calls `attach_data` passing it the contents of that file.

However, there are some issues with attachments in Xeroizer at the moment:

1. Attachments can't be included with the invoice. They will show up under "RELATED FILES" in Xero, but will not be listed under "TO INCLUDE WITH INVOICE".
2. Filenames with spaces will not work due to some escaping issues

I'll work on patches for both issues as soon as I get a chance.

**Comment by maxime-d** [comment]
Thanks, I solved the issue with your comment `Filenames with spaces will not work due to some escaping issues` !

**Comment by nnc** [comment]
Cool. Glad it worked out for you.

Including attachments with invoices was addressed in #241, while filename escaping is still not to Xero's liking, but that should be a separate ticket.

@maxime-d you should probably close this ticket now.
